src/openassetio-core/include/openassetio/Context.hpp

```diff
@@ -41,10 +41,32 @@ class OPENASSETIO_CORE_EXPORT Context final {
   /**
    * @name Access Pattern
    */
-  enum class Access { kRead, kReadMultiple, kWrite, kWriteMultiple, kUnknown };
+  enum class Access {
+    /**
+     * Host intends to read data
+     */
+    kRead,
+    /**
+     * Host intends to write data. This should be the default choice for
+     * write, unless the conditions for @ref Access.kCreateRelated are
+     * met.
+     */
+    kWrite,
+    /**
+     * Hosts intends to write related data against a reference to
+     * another entity. This is a specialization of `kWrite`, and should
+     * be used when the host knows up front that it wishes to to publish
+     * a new related entity, and not an update to an existing entity.
+     * The canonical motivating example for this is that host may know
+     * it wishes to publish a new subfolder inside an existing folder,
+     * and not an update to said existing folder.
```

Comment:
[fix] Attempting to align with the `kWrite` suggestion

```suggestion
     * Hosts intends to write data for a new entity in relation to another.
       During publishing this must be used whenever the entity reference 
       points to an existing entity, and the intention is to create a  new, 
       related entity instead of updating the target. For example, when 
       programmatically creating new  entities under an existing parent
       collection, or the publishing of the components of a structured
       asset based on a single root entity reference.
```

(I was just trying to avoid 'folder' due to the file-system connotations).

Reply:
Yeah, avoiding folder is wise
https://github.com/OpenAssetIO/OpenAssetIO/pull/1036/commits/08505a52d4e9ac7232f457e4b21c853841ffb416